udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -53,14 +53,18 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     Query query = new Query("CommentSingle").addSort("timestamp", SortDirection.DESCENDING);
 
     PreparedQuery results = datastore.prepare(query);
-
+    int limiter = getCommentCount(request);
```

Comment:
I would suggest getting the limit before constructing the query, and then using it in a LIMIT clause in the query. SQL can take care of limiting results for you so you don't need to manually limit your output below.

Reply:
I'm only seeing it possible online to limit a sorted query after the initial query through FetchOptions as is being done.